examples/forum/TUTORIAL.md

```diff
@@ -694,6 +680,33 @@ comment on function forum_example.current_person() is 'Gets the person who was i
 
 This is a simple function that we can use in PostGraphile or our database to get the person who is currently executing the query — by means of the token in the request header. The one new concept here is `current_setting('jwt.claims.person_id', true)::integer`. As we discussed before, PostGraphile will serialize your JWT to the database in the form of transaction local settings. Using the `current_setting` function is how we access those settings. Also note that we cast the value to an integer with `::integer`. This is because the Postgres `current_setting` function will always return a string, if you need another data type, you will likely need to cast to that data type.
 
+### Updating Passwords
+
+Now that we can identify the current user, we might want to use that information to make functions that are relevant only to the current user, such as returning a set of data only relevant to that user or, as our title suggests, changing the current user's password.
+
+```sql
+create function forum_example.change_password(current_password text, new_password text) 
+returns boolean as $$
+declare
+  current_person forum_example.person;
+begin
+  current_person := forum_example.current_person();
+  if exists (select 1 from forum_example_private.person_account where person_account.person_id = current_person.id and person_account.password_hash = crypt($1, person_account.password_hash)) 
+  then
+    update forum_example_private.person_account set password_hash = crypt($2, gen_salt('bf')) where person_account.person_id = current_person.id; 
+    return true;
+  else 
+    return false;
+  end if;
+end;
+$$ language plpgsql strict security definer;
+```
```

Comment:
This could also be accomplished with SQL; something like:

```sql
create function forum_example.change_password(current_password text, new_password text) 
returns boolean as $$
  with updated as (
    update forum_example_private.person_account
      set password_hash = crypt($2, gen_salt('bf'))
      where person_account.person_id = (forum_example.current_person()).id
      and person_account.password_hash = crypt($1, person_account.password_hash)
      returning *;
  )
  select exists (select 1 from updated);
end;
$$ language plpgsql strict security definer;
```

(untested)